torch/backends/cudnn/__init__.py

```diff
@@ -166,256 +183,257 @@ def flags(enabled=False, benchmark=False, deterministic=False, verbose=False):
             set_flags(orig_flags[0], orig_flags[1], orig_flags[2], orig_flags[3])
 
 
-class CuDNNHandle:
-    def __init__(self):
-        ptr = ctypes.c_void_p()
-        check_error(lib.cudnnCreate(ctypes.byref(ptr)))
-        self._as_parameter_ = ptr
-
-    def __del__(self):
-        check_error(lib.cudnnDestroy(self))
-
+if cuda is not None:
```

Comment:
That's true. The error code will itself run into an error since it tries to use the cuDNN api:
```
    def get_error_string(status):
        return lib.cudnnGetErrorString(status)
```
If that's not an issue from a user perspective, I can remove the guard.

Reply:
Yeah I agree it's not ideal. But I think there is not much difference on the error message by not defining these python classes (Like users will see "`get_error_string` not found in the cudnn module" vs. "`cudnnGetErrorString` symbol not found in lib").
One solution would be us defining a shim in torch library in c++ (inside which we switch to use cudnnGetErrorString or miopenGetErrorString based on the `__HIP_PLATFORM_HCC__` define).

Reply:
Okay, in that case, since we're not planning to currently add MIOpen support for any of the classes/methods in `__init__.py` which call these cuDNN lib functions, I'll just remove the guard for now. If/when we decide to add MIOpen support, we can add the shim in C++.